samples/nextjs/scripts/generate-config.ts

```diff
@@ -33,17 +33,30 @@ export function generateConfig(configOverrides?: { [key: string]: string }): voi
 
   // The GraphQL endpoint is an example of making a _computed_ config setting
   // based on other config settings.
-  addGraphQLConfig(config);
-
-  const configText = `/* eslint-disable */
+  const computedConfig: { [key: string]: string } = {};
+  computedConfig.graphQLEndpoint =
+    '`${config.sitecoreApiHost}${config.graphQLEndpointPath}?sc_apikey=${config.sitecoreApiKey}`';
+
+  // Create an empty object defining all our configuration properties, so module export has proper typing
+  const configProps = Object.assign({}, config, computedConfig);
+  Object.keys(configProps).forEach((prop) => (configProps[prop] = ''));
+  let configText = `/* eslint-disable */
 // Do not edit this file, it is auto-generated at build time!
 // See scripts/bootstrap.ts to modify the generation of this file.
-module.exports = ${JSON.stringify(config, null, 2)};`;
+var config = ${JSON.stringify(configProps, null, 2)};\n`;
+
+  // Set base configuration values, allowing override with environment variables
+  Object.keys(config).forEach((prop) => {
+    configText += `config.${prop} = process.env.${prop} || "${config[prop]}",\n`;
```

Comment:
Any slick way we can convert the prop names to follow environment variable naming syntax (i.e. "SITECORE_API_KEY" vs "sitecoreApiKey")?

Reply:
Done!